### ISSUE TITLE
MySQL 5.7 image failed while attempting to check config

### ISSUE THREAD

**Issue by Enqueued** [docker-library/mysql#873]
### Introduction:
---
Hello, I am running Fedora 36 and trying to run the docker image for MySql 5.7
I had been using an already built container for some time but recently when I tried to start the image I am met with an error saying that it failed when checking config.
I have been able to successfully build a docker image for MySQL 8 but I need 5.7 for legacy support.

I have read issue #840 about it failing due to a bug in Fedora 35 but it ends at that point.
Has there been any way to work around this or has it be completed as of F36? 

### Docker Run Command:
---
 docker run -p 3320:3306 --name dev-mysql -v /home/path/to/logs/sql_docker_conf:/etc/mysql/conf.d -e MYSQL_ROOT_PASSWORD=super-secret-pass -d 74a6a7ec181f 

### Docker Logs:
---
 2022-07-01 17:50:59+00:00 [Note] [Entrypoint]: Entrypoint script for MySQL Server 5.7.38-1debian10 started.
2022-07-01 17:51:08+00:00 **[ERROR] [Entrypoint]: mysqld failed while attempting to check config
	command was: mysqld --verbose --help --log-bin-index=/tmp/tmp.wAi7qWrgpp** 

### Fetch:
---
kernel ~ 5.18.7-200.fc36.x86_64
os ~ Fedora Linux 36 (Workstation Edition)   
arch ~ x86_64
de/wm ~ GNOME   
cpu ~ AMD Ryzen 7 4700U with Radeon Graphics 	2.00GHz@4.0
gpu ~ AMD/ATI Renoir (rev c2)
ram ~ 15246 MiB   swap ~ 8191 MiB
term ~ kitty   shell ~ zsh

**Comment by wglambert** [comment]
You'll want to look at the main bug report for it over at https://github.com/containerd/containerd/issues/6707
Looks like there's some workarounds you could try with setting `LimitNOFILE=1048576` https://github.com/containerd/containerd/issues/6707#issuecomment-1101951667

But there isn't anything we can reasonably do at this level to resolve it since it's not an issue with the image itself so going to close as a duplicate of https://github.com/docker-library/mysql/issues/840

**Comment by JCenatus** [comment]
Hello,

I know this post is old, nevertheless, for those who come across it in the future like me, this solution allows resolving the issue without intervening in the host system configuration:

```
---
version: '3.6'
services:
  mysql:
    image: mysql:5.7
    ulimits:
      nofile: # Fix memory leak issue on some systems when LimitCORE=infinity (containerd)
        soft: 1048576
        hard: 1048576
```


This is a configuration for Compose (Docker plugin). However, you can use the equivalent, such as `docker run mysql:5.7 --ulimit nofile=1048576:1048576`.

For your information, I encountered a similar issue with the `library/elasticsearch:2.4-alpine` image.

I use Fedora Workstation Silverblue 39

**Comment by LorenzoMorelli** [comment]
> Hello,
> 
> I know this post is old, nevertheless, for those who come across it in the future like me, this solution allows resolving the issue without intervening in the host system configuration:
> 
> ```
> ---
> version: '3.6'
> services:
>   mysql:
>     image: mysql:5.7
>     ulimits:
>       nofile: # Fix memory leak issue on some systems when LimitCORE=infinity (containerd)
>         soft: 1048576
>         hard: 1048576
> ```
> 
> This is a configuration for Compose (Docker plugin). However, you can use the equivalent, such as `docker run mysql:5.7 --ulimit nofile=1048576:1048576`.
> 
> For your information, I encountered a similar issue with the `library/elasticsearch:2.4-alpine` image.
> 
> I use Fedora Workstation Silverblue 39

This workaround works also on my Archlinux. Thank you!

**Comment by cschindlbeck** [comment]
> Hello,
> 
> I know this post is old, nevertheless, for those who come across it in the future like me, this solution allows resolving the issue without intervening in the host system configuration:
> 
> ```
> ---
> version: '3.6'
> services:
>   mysql:
>     image: mysql:5.7
>     ulimits:
>       nofile: # Fix memory leak issue on some systems when LimitCORE=infinity (containerd)
>         soft: 1048576
>         hard: 1048576
> ```
> 
> This is a configuration for Compose (Docker plugin). However, you can use the equivalent, such as `docker run mysql:5.7 --ulimit nofile=1048576:1048576`.
> 
> For your information, I encountered a similar issue with the `library/elasticsearch:2.4-alpine` image.
> 
> I use Fedora Workstation Silverblue 39

Works on Arch too

**Comment by phonkd** [comment]
I have no idea how but this fixed it ty

**Comment by nickdollimount** [comment]
I wanted to add a comment to this in case people search for it with regards to Ubuntu Server 24.10. I just upgraded to server to 24.10 last night and noticed today that a couple sites I host were broken, both issues connecting to the database. Checked BTOP and the memory and CPU were going haywire. Checked the docker log in both and they both had the same error message as this issue. @JeanLuX comment above fixed this for me.

I'm not sure if this is relevant but one instance is 5.7 and the other is 8.0.3, both with the same issue.
